react/features/prejoin/components/Prejoin.native.tsx

```diff
@@ -0,0 +1,132 @@
+}
+
+const Prejoin: ({ navigation }: Props) => JSX.Element = ({ navigation }: Props) => {
+    const dispatch = useDispatch();
+    const { t } = useTranslation();
+    const aspectRatio = useSelector(
+        (state: any) => state['features/base/responsive-ui']?.aspectRatio
+    );
+    const store = useStore();
+    const state = store.getState();
+    const localParticipant = useSelector(state => getLocalParticipant(state));
+    const participantName = localParticipant?.name;
+    const [ displayName, setDisplayName ]
+        = useState(participantName || '');
+    const onChangeDisplayName = useCallback(event => {
+        const fieldValue = getFieldValue(event);
+
+        setDisplayName(fieldValue);
+        dispatch(updateSettings({
+            displayName: fieldValue
+        }));
+    }, [ displayName ]);
+
+    const onJoin = useCallback(() => {
+        dispatch(connect());
+        navigateRoot(screen.conference.root);
+    }, [ dispatch ]);
+
+    const goBack = useCallback(() => {
+        goBackToRoot(state, dispatch);
+    }, [ dispatch ]);
+
+    let contentWrapperStyles;
+    let contentContainerStyles;
+    let largeVideoContainerStyles;
+    let toolboxContainerStyles;
+
+    if (aspectRatio === ASPECT_RATIO_NARROW) {
+        contentWrapperStyles = styles.contentWrapper;
+        contentContainerStyles = styles.contentContainer;
+        largeVideoContainerStyles = styles.largeVideoContainer;
+        toolboxContainerStyles = styles.toolboxContainer;
+    } else {
+        contentWrapperStyles = styles.contentWrapperWide;
+        contentContainerStyles = styles.contentContainerWide;
+        largeVideoContainerStyles = styles.largeVideoContainerWide;
+        toolboxContainerStyles = styles.toolboxContainerWide;
+    }
+
+    const headerLeft = useCallback(() => {
+        if (Platform.OS === 'ios') {
+            return (
+                <HeaderNavigationButton
+                    label = { t('dialog.close') }
+                    onPress = { goBack } />
+            );
+        }
+
+        return (
+            <HeaderNavigationButton
+                onPress = { goBack }
+                src = { IconClose } />
+        );
+    }, []);
+
+    useLayoutEffect(() => {
+        navigation.setOptions({
+            headerLeft
+        });
+    }, [ navigation ]);
+
+    return (
+        <JitsiScreen
+            safeAreaInsets = { [ 'right' ] }
+            style = { contentWrapperStyles }>
+            <View style = { largeVideoContainerStyles }>
+                <LargeVideo />
+            </View>
+            <View style = { contentContainerStyles }>
+                <View style = { styles.formWrapper }>
+                    <TextInput
+                        onChangeText = { onChangeDisplayName }
+                        placeholder = { t('dialog.enterDisplayName') }
+                        style = { styles.field }
+                        value = { displayName } />
+                    <TouchableOpacity
```

Comment:
We should have the "join in low bandwidth mode" secondary button after this one, which basically dispatches setAudioOnly(true) and then joins.